xarray/core/dataarray.py

```diff
@@ -3188,17 +3188,20 @@ def differentiate(
         return self._from_temp_dataset(ds)
 
     def integrate(
-        self, dim: Union[Hashable, Sequence[Hashable]], datetime_unit: str = None
+        self,
+        coord: Union[Hashable, Sequence[Hashable]],
+        dim: Union[Hashable, Sequence[Hashable]] = None,
+        datetime_unit: str = None,
```

Comment:
`dim` should be a keyword-only parameter until it is removed. Otherwise, this would be possible:
```python
da.integrate("x", "y", "ns")
```
so:
```suggestion
        coord: Union[Hashable, Sequence[Hashable]] = None,
        datetime_unit: str = None,
        *,
        dim: Union[Hashable, Sequence[Hashable]] = None,
```
However, this means that you have to guard against specifying both (or none) of them